[#62217] allow custom fields that are not enabled in project

Kharonus opened the pull request:
# Ticket
[OP#62217](https://community.openproject.org/work_packages/62217)

# What are you trying to accomplish?
- allow custom fields in generated subjects that are not enabled in project

# What approach did you choose and why?
- rescue when sending custom field methods

Kharonus commented:
@mereghost we need to talk about this one :D I know, that you will have an opinion on it.

mereghost commented:
Nice. Do you remember if the custom field code makes changes to `respond_to_missing?`? If so we could just return nil when `context.responds_to?(:method_name)` is false.

Cheaper than the exception.

mereghost reviewed (approved):
This works well enough. But see comment in PR. =)

Kharonus committed:
[#62217] allow custom fields that are not enabled in project

- https://community.openproject.org/work_packages/62217
- rescue when sending custom field methods

Kharonus commented:
Changed to `respond_to?` and added a unit test
